### DISCUSSION TITLE
PSB file with only one layer (background layer)

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#6109] [Help] [answered]
### ImageMagick version

7.1.0-62 Q16-HDRI x64

### Operating system, version and so on

Win 10 Pro 64 22H2

### Description

Hello, I don't know the exact name in English but I want to convert TIFF or PNG files to PSB flattened to the background layer.

Since the files are technically processed further, they must not contain any further layers.

Unfortunately I can't find any documentation anywhere. I've tried everything obvious like `-flatten`, `-coalesce`, `-layer merge`, but whenever I check the file in Photoshop <ins>I can still merge the layers onto the background layer</ins>. If i create the file directly in PS and open it _there are no additional layers_.

**Unfortunately, that must not be the case.**

So, please help me in this line:

```batch
magick in.tiff -OPTION out_as_merged_layers.psb
```

`in.tiff` in this case can be any BigTIFF or PNG. Maybe I'm just missing the right name to look for the option.

Thank you

**Comment by fmw42** [comment]
Post your Tiff file.  Try this

`magick in.tiff +adjoin -coalesce -background none -flatten out_as_merged_layers.psb`

**Reply by ghost** [reply]
Thank you @fmw42 but this gives the same output.

My files are several gigabytes in size because they are gigapixel images. To post it:

- where can i post it
- which size makes sense? I will try to reduce as small as possible containing the layers.
- not the PSB?

Because we are talking about BigTIFF (and PNG) i must test if there is a difference in layers.

I need the gigapixel sizes.

**Reply by ghost** [reply]
Ok, resized @fmw42 

Where can i upload those "origin" files for you?

**Reply by fmw42** [reply]
Can you upload a smaller tiff file for testing that has similar layers?  If the small file works, perhaps your larger one will work also.

You can zip it an post here or post to some free hosting service and put the URL here.

**Reply by ghost** [reply]
@fmw42 Thank you

[files.zip](https://github.com/ImageMagick/ImageMagick/files/10834614/files.zip)

**Comment by fmw42** [comment]
The tiff file you posted has only one layer. I am confused. I thought you said you had a tiff with multiple layers. Please clarify.

Also all your files show only one layer and they all look the same. Except for png2.psb, which has a black preview, but the file itself looks the same as the others when I view them all in Photoshop.

**Reply by ghost** [reply]
The PSB (see zip) has more than one layer after `magick x.tif y.psb`

In Photoshop you can now right-click merge to bottom layer ("don´t know the english words")

I want to tell ImageMagick to create a PSB file without additional layers.

Can you open the PSB in Photoshop please? In the layers window you can now merge to background - this is only because there are additional layers.

**Reply by ghost** [reply]
@fmw42 

Here is a new zip: [nolayers.zip](https://github.com/ImageMagick/ImageMagick/files/10834871/nolayers.zip)

In Photoshop i created an empty white TIFF - no additional layer included

now `magick empty.tif magick.psb` (magick.psb) produces additional layer

instead exporting by Photoshop (photoshop.psb) - no additional layer

As you can see, too, the photoshop PSB of the same file has 33KB while magick.psb has 911KB.

_Edit: ok, 33KB seems to be compressed. The PSB should be a bit bigger than TIFF. The emtpy.tif has 473KB, but here you can see the double layer file size. Exporting in PS with or without the additional layer gives a double of the size..._

**Reply by fmw42** [reply]
I am still confused.  All three of your files magick.psb empty.tif photoshop.psb have only one layer. All 3 look the same in Photoshop.  What is different about them?

Also wk1.tiff has only one layer in Photoshop.

**Reply by ghost** [reply]
This is empty.tif:

![image](https://user-images.githubusercontent.com/126161988/221442138-8049319e-b195-4472-85f9-1bf70109fe5c.png)

This is photoshop.psb:

![image](https://user-images.githubusercontent.com/126161988/221442187-c8a229e8-b613-48ff-9381-daafc142d178.png)

_Both are not containing additional layers_

This is magick.psb, you can see the **inner border** (in the thumbnail) , the naming **L1**, and the last item in context menu:

![image](https://user-images.githubusercontent.com/126161988/221442293-9507032b-9b53-43e2-a025-5b64b7b87208.png)

**Reply by ghost** [reply]
This is technically a difference (to further processing) and in size as all layers are double in size because it is flat / linear RGB in PSB files, roughly like raw photo files...

**Reply by ghost** [reply]
It says "Merge to Background Layer" ("Auf Hintergrundebene reduzieren") - so there is an additional layer or background layer added.

**Comment by fmw42** [comment]
Sorry, I only see one layer

![Screen Shot 2023-02-26 at 3 32 07 PM](https://user-images.githubusercontent.com/13458218/221444149-c2f6cbcb-22a8-451b-8a52-e391b092ce99.png)

Why are you trying to merge to Background Layer if you only have one layer?

**Reply by ghost** [reply]
Problem is the PSB files generated by ImageMagick. The TIFFs are not from ImageMagick. You please have to look at the generated PSB, they are the problem.

ImageMagick adds additional Layer to PSB.

**Reply by ghost** [reply]
It seems to be that ImageMagick creates a background layer and a visibility layer. Instead of origin 16GB, my files have 32 GB and cannot be further processed

**Comment by ghost** [comment]
That is why i am looking for the option to create a _flatten_ PSB

**Comment by fmw42** [comment]
I am not an expert on this and an IM developer may have to comment. But every file you have sent me including the PSB files have only one layer. The layer name has changed from L1 to Background on some of these files, but the images are still the same and there is only one layer.  I do not think IM knows about layer names and just names them Background.  But the IM developer may correct me about this.

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/126161988/221654677-1af4ae76-d30f-490e-b99c-ff4b370be0e6.png)

I will file an issue.

**Comment by ghost** [comment]
https://github.com/ImageMagick/ImageMagick/issues/6113

**Comment by ghost** [comment]
@fmw42 That wasn't for you anymore, that's for others who are also asking the question

**Comment by dlemstra** [comment, marked as answer]
What happens when you add `-define psd:write-layers=false`? Does that give you the expected output?

**Comment by fmw42** [comment]
I think that works for me. I get one layer that is 2 MiB, similar to tiff

```
magick original.tif -define psd:write-layers=false original.psb
magick identify original.psb original.tif
original.psb PSB 600x600 600x600+0+0 16-bit sRGB 2.06562MiB 0.000u 0:00.002
original.tif TIFF 600x600 600x600+0+0 16-bit sRGB 2.07992MiB 0.000u 0:00.001
```

**Comment by ghost** [comment]
Thank you very much :-)

**Comment by fmw42** [comment]
Sorry for my misunderstanding of the issue.  And thanks to dlemstra for the reference to the -define that solved it.

**Comment by ghost** [comment]
@fmw42 I have already noticed you several times, madness, how much, often and quickly you help everyone here!

Thank you for your quick responses
